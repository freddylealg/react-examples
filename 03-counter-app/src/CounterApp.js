import PropTypes from 'prop-types';
import { useState } from 'react';

const CounterApp = ({value=0}) => {
    
    const [count, setValue] = useState(value);

    // cambio directo
    const handleAdd = () => {
        setValue(count + 1);
    }

    const handleReset = () => {
        setValue( value );
    }

    // cambio con funciÃ³n
    const handleSubtrat = () => {
        setValue( (count) => count - 1 );
    }

    return (
        <div>
        <h1>Counter App</h1>
        <h2>Count: {count}</h2>

        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleReset }>Reset</button>
        <button onClick={ handleSubtrat }>-1</button>
        </div>
    );
}

CounterApp.prototype = {
    value: PropTypes.number
}

export default CounterApp;