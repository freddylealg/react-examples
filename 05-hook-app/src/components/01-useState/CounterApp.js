import React, { useState } from 'react'
import './counter.css';

export const CounterApp = () => {

    const [counterObj, setCounter] = useState({
        count1: 0,
        count2: 10,
        count3: 20,
        count4: 30,
    });

    const {count1, count2} = counterObj;

    return (
        <>
            <h1>Counter1 { count1 }</h1>
            <h1>Counter2 { count2 }</h1>
            <hr />
            <button className="btn btn-primary" 
                onClick={
                    () => {setCounter( {...counterObj, count1: count1 + 1} )}
                } 
            >+1</button>
        </>
    )
}
