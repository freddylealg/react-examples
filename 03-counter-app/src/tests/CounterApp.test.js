import React from 'react';
import { shallow } from 'enzyme'
import CounterApp from "../CounterApp";
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';


describe('Pruebas del <CounterApp />', () => {

    let wrapper = shallow( <CounterApp /> );

    beforeEach( () => { 
        wrapper = shallow( <CounterApp /> );
    });

    test('Debe mostrar CounterApp correctamente', () => {
        expect( toJson(wrapper) ).toMatchSnapshot();
    });

    test('Prueba del valor por defecto del CounterApp', () => {
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toBe( 'Count: 0' );
    });

    test('Prueba del valor por defecto del CounterApp', () => {
        const value = 100;
        const wrapper = shallow( <CounterApp value={value} /> );
        
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toBe( `Count: ${value}` );
    });

    test('Incrementar con el botón +1', () => {
        const btn1 = wrapper.find('button').at(0).simulate('click');
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toBe( `Count: 1` );
    });

    test('Decrementar con el botón -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toBe( `Count: -1` );
    });

    test('Botón de reset', () => {
        const value = 100;
        const wrapper = shallow( <CounterApp value={value} /> );

        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toBe( `Count: ${value}` );
    });

});