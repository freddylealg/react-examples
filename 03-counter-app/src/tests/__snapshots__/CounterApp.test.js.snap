// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <CounterApp /> Debe mostrar CounterApp correctamente 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    Count: 
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
